api/views/curriculum.py

```diff
@@ -0,0 +1,161 @@
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  A copy of this license - GNU General Public License - is available
+#  at the root of the source code of this program.  If not,
+#  see http://www.gnu.org/licenses/.
+#
+# ##############################################################################
+from functools import partial
+from typing import Optional
+
+from django.db.models import Exists, OuterRef
+from django.utils.translation import gettext as _
+from rest_framework import status
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.generics import get_object_or_404, RetrieveAPIView
+from rest_framework.mixins import UpdateModelMixin
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from admission.api import serializers
+from admission.api.permissions import IsSelfPersonTabOrTabPermission
+from admission.api.schema import ResponseSpecificSchema
+from admission.api.views.mixins import PersonRelatedMixin, PersonRelatedSchema
+from admission.contrib.models import DoctorateAdmission
+from osis_profile.models import Experience
+from osis_role.contrib.views import APIPermissionRequiredMixin
+
+
+class CurriculumExperienceSchema(ResponseSpecificSchema, PersonRelatedSchema):
+    operation_id_base = "_curriculum_experience"
+
+    serializer_mapping = {
+        "GET": serializers.ExperienceOutputSerializer,
+        "PUT": (
+            serializers.ExperienceInputSerializer,
+            serializers.ExperienceOutputSerializer,
+        ),
+        "POST": (
+            serializers.ExperienceInputSerializer,
+            serializers.ExperienceOutputSerializer,
+        ),
+        "DELETE": (),
+    }
+
+
+class CurriculumExperienceView(PersonRelatedMixin, APIPermissionRequiredMixin, APIView):
+    schema = CurriculumExperienceSchema()
+    permission_classes = [
+        partial(IsSelfPersonTabOrTabPermission, permission_suffix="curriculum")
+    ]
+    name = "curriculum"
+
+    def get_queryset(self):
+        """Return the list of experiences of the person."""
+        return (
+            Experience.objects.filter(curriculum_year__person=self.get_object())
+            .annotate(
+                is_valuated=Exists(
+                    DoctorateAdmission.valuated_experiences.through.objects.filter(
+                        experience_id=OuterRef("pk"),
+                    )
+                )
+            )
+            .order_by("curriculum_year__academic__year")
+        )
+
+    def get_experience(self) -> Optional[Experience]:
+        """Get the current experience from the uuid."""
```

Comment:
from the pk.

Reply:
Sorry ! Comme sébastien me faisait remarquer; il est préférable de communiquer via leurs uuids et non leur pk; pourrais-tuc changer cela?

Reply:
D'accord c'est noté. J'ai ajouté cette nouvelle propriété dans le modèle que j'utilise maintenant côté backoffice et frontoffice.